src/style/color.rs

```diff
@@ -153,6 +153,43 @@ impl serde::Serialize for Color {
 
 #[cfg(feature = "serde")]
 impl<'de> serde::Deserialize<'de> for Color {
+    /// This is used to deserialize a value into Color via serde.
+    ///
+    /// This implementation uses the `FromStr` trait to deserialize strings, so named colours, RGB,
+    /// and indexed values are able to be deserialized. In addition, values that were produced by
+    /// the the older serialization implementation of Color are also able to be deserialized.
+    ///
+    /// See the [`Color`] documentation for more information on color names.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ratatui::prelude::*;
+    ///
+    /// #[derive(serde::Deserialize)]
+    /// struct Theme {
+    ///     color: Color,
+    /// }
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"bright-white\"}").unwrap();
```

Comment:
- Use a raw string to make this easier to read `r#"{ "color": "bright-white" }"#
- Avoid using unwrap by adding wrapping the example in a method that returns Result (this can be hidden from from the docs using `#`):
- Remove any unnecessary turbo fish / casts...

E.g.:

```rust
/// # fn get_theme() -> Result<(), ParseError> {
/// let theme: Theme = serde_json::from_str(r#"{ "color": "bright-white" }"#)?;
/// assert...
/// # Ok(())
/// # }
```